### DISCUSSION TITLE
Does  this DDS project support java?

### DISCUSSION THREAD

**Discussion by huangjie84** [eProsima/Fast-DDS#5022] [Q&A]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

i find the fast dds gen project has support for java(JNI).but i can't find any examples.
is there some demo has passed tests?
really appreciated

### Current behavior

this is no examples

### Steps to reproduce

1.install fastDDS
2.download fastDDSGen
3.write a simple .idl file
4.use gen project to make C++ & java code   [ error happend in this step]
5. run it 

### Fast DDS version/commit

master

### Platform/Architecture

Windows 10 Visual Studio 2019

### Transport layer

Default configuration, UDPv4 & SHM

**Comment by Mario-DL** [comment]
Hi @huangjie84,

We do not still support `Java`. It is true that `Fast DDS Gen` can generate a kind of bindings for Java (in fact only for the type, not the `Typesupport`) but that option is going to be removed since it is not actively maintained right now. You might consider contacting [eProsima's commercial support team](https://github.com/eProsima/Fast-DDS/tree/master#getting-help) for premium support options if this is a need for your project.

Moving this thread to the corresponding discussion forum.
